File: exam/string/parenthesis/parenthesis.cpp
#include <iostream>
using namespace std;

int find(string data)
{
    int opening = 0, closing = 0;

    for (char c : data)
    {
        if (c == 40)
        {
            opening++;
        }
        else
        {
            if (opening > 0)
            {
                opening--;
            }
            else
            {
                closing++;
            }
        }
    }

    return opening + closing;
}
int main()
{
    string data1 = "(()))";
    string data2 = "))((";

    cout << "Result 1 is " << find(data1) << endl;
    cout << "Result 2 is " << find(data2) << endl;
}

File: exam/string/parenthesis/parenthesis.dart
void main() {
  String data1 = "(()))";
  String data2 = "))((";

  print("(".codeUnits);
  print(")".codeUnits);

  List<String> input1 = data1.split("").toList();
  List<String> input2 = data2.split("").toList();

  print("Result 1 : ${find(input1)}");
  print("Result 2 : ${find(input2)}");
}

int find(List<String> data) {
  int open = 0;
  int close = 0;
  for (var i = 0; i < data.length; i++) {
    if (data[i] == "(") {
      open++;
    } else {
      if (open > 0)
        open--;
      else
        close++;
    }
  }

  return open + close;
}
